### PR TITLE
[MM-14593] Add password type for Interactive Dialogs

### PR DESCRIPTION
**Pull request by aaronrothschild**
- Add textarea SUB-TYPE to interactive dialogs

https://github.com/mattermost/mattermost-webapp/pull/3057

### CODE REVIEW

File: source/developer/interactive-dialogs.rst [hunk 1]
```diff
@@ -130,13 +130,15 @@ The list of supported fields is the same as for the ``textarea`` type element.
...
     "type", "String", "Set this value to ``textarea`` for a textarea element."
-    "subtype", "String", "(Optional) One of ``text``, ``email``, ``number``, ``tel``, or ``url``. Default is ``text``. Use this to set which keypad is presented to users on mobile when entering the field."
+    "subtype", "String", "(Optional) One of ``text``, ``email``, ``number``, ``password``,``tel``, or ``url``. Default is ``text``. Use this to set which keypad is presented to users on mobile when entering the field."
     "min_length", "Integer", "(Optional) Minimum input length allowed for an element. Default is 0."
     "max_length", "Integer", "(Optional) Maximum input length allowed for an element. Default is 3,000."
     "optional", "Boolean", "(Optional) Set to true if this form element is not required. Default is ``false``."
     "help_text", "String", "(Optional) Set help text for this form element. Maximum 150 characters."
     "default", "String", "(Optional) Set a default value for this form element. Maximum 3,000 characters."
     "placeholder", "String", "(Optional) A string displayed to help guide users in completing the element. Maximum 3,000 characters."
+
```
Comment (hanzei) [inline source/developer/interactive-dialogs.rst]: Why did you add these lines?

File: source/developer/interactive-dialogs.rst [hunk 2]
```diff
@@ -130,13 +130,15 @@ The list of supported fields is the same as for the ``textarea`` type element.
     "display_name", "String", "Display name of the field shown to the user in the dialog. Maximum 24 characters."
     "name", "String", "Name of the field element used by the integration. Maximum 300 characters. You should use unique “name” fields in the same dialog."
     "type", "String", "Set this value to ``textarea`` for a textarea element."
-    "subtype", "String", "(Optional) One of ``text``, ``email``, ``number``, ``tel``, or ``url``. Default is ``text``. Use this to set which keypad is presented to users on mobile when entering the field."
+    "subtype", "String", "(Optional) One of ``text``, ``email``, ``number``, ``password``,``tel``, or ``url``. Default is ``text``. Use this to set which keypad is presented to users on mobile when entering the field."
```
Comment (hanzei) [inline source/developer/interactive-dialogs.rst]: I would like to also see the required Mattermost version for this documented. But I'm unsure how to document this properly. Maybe we add a short section for each `subtype` with a screenshot? 1/5

**Comment by amyblais** [comment]
@aaronrothschild Can you help finish this doc?

File: source/developer/interactive-dialogs.rst (hunk 2 above)
Comment (aaronrothschild) [reply source/developer/interactive-dialogs.rst]: I added the 5.14 requirement in the text - it's on Master already:  https://github.com/mattermost/docs/blob/master/source/developer/interactive-dialogs.rst

File: source/developer/interactive-dialogs.rst (hunk 1 above)
Comment (aaronrothschild) [reply source/developer/interactive-dialogs.rst]: Accidentally.

**Comment by aaronrothschild** [comment]
Sorry Amy, this is an old branch and I updated the docs through one of my "squished" updates for 5.14 I think.  All of Hanzei's comments were incorporated into: https://github.com/mattermost/docs/blob/master/source/developer/interactive-dialogs.rst

**Comment by hanzei** [comment]
@aaronrothschild Password for dialogs are only available from `v5.16` onward. Would you please fix this?
